docs/docker_host_ssh.md

```diff
@@ -0,0 +1,116 @@
+Reuse docker host SSH daemon
+============================
+
+It is possible to use the SSH daemon that runs on the docker host instead of using a separate SSH port for GitLab.
+
+## Setup
+
+Create the system `git` user if necessary (make sure it has a working shell):
+
+```bash
+useradd --system git --home-dir /srv/docker/gitlab/data
+```
+
+The home directory must point to GitLab data directory, as it takes `.ssh/authorized_keys` from there.
+
+Now create `gitlab-shell` home directory at the same location as in the container:
+
+```bash
+mkdir -p /home/git/gitlab-shell/bin
```

Comment:
@IlyaSemenov: Ah, I wasn't asking about permissions though, but about the directory/file paths in the guide, which I couldn't reconcile with what I was seeing locally. But it was just me being confused and thinking this guide was for the official omnibus image.

(The permission problem I solved by matching the git uid/gid on the host with those hard-coded in the official container, instead of chmod:ing, since we don't want to run OpenSSH with StrictModes off.)